### PR TITLE
Optimize commons files - configurable.rb & default.rb

### PR DESCRIPTION
**Pull request by sarangan12**
Removed the process of copying configurable.rb and default.rb to all gems. Moved the files to MsRestAzure and optimized the code.

@devigned @veronicagg @salameer Please review and approve.

### CODE REVIEW

**Comment by devigned** [comment]
@sarangan12 the CI is failing with all but the latest runtime due to the changes only being in the latest runtime, right?

**Comment by sarangan12** [comment]
@devigned Yes. The latest runtime has the required code changes while the published ones don't

**Comment by sarangan12** [comment]
@devigned Yes. The latest runtime has the required code changes while the published ones don't

**Comment by sarangan12** [comment]
@devigned Yes. The latest runtime has the required code changes while the published ones don't

**Comment by sarangan12** [comment]
@devigned Yes. The latest runtime has the required code changes while the published ones don't

File: management/azure_mgmt_network/spec/2017-09-01/load_balancers_spec.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@
 
     load_balancing_rule = LoadBalancingRule.new
     load_balancing_rule.name = load_balancing_rule_name
-    frontend_ip_configuration_sub_resource = MsRestAzure::SubResource.new
```
Comment (veronicagg) [inline management/azure_mgmt_network/spec/2017-09-01/load_balancers_spec.rb]: this change is not related to the PR, right? I believe it's related to https://github.com/Azure/azure-sdk-for-ruby/pull/1106, where should have this been caught when the other PR changes happened?
